### ISSUE TITLE
Automate formatting and check it in CI

### ISSUE THREAD

**Issue by tbroyer** [SORMAS-Foundation/SORMAS-Project#3543]
Currently, formatting the code (including ensuring license headers are correctly added/updated) requires manually configuring your IDE, which itself possibly requires installing additional plugins.

There's also no process for checking that files are properly formatted, so when formatting a file you might end up actually fixing bad formatting from a previous commit, leading to changes that are unrelated to the bugfix/feature being worked on thus making code reviews (or committing, if the develop decides to ignore those unrelated reformatting changes) harder.

Automating all of this would raise the overall project quality and streamline on-boarding of new contributors.

Fortunately, this is relatively easy to do through [Spotless](https://github.com/diffplug/spotless) and their Maven plugin.

I did make those changes locally as I didn't want to install the IntelliJ IDEA plugin allowing me to configure it, and then take time to configure both that plugin and the license headers; and I'd happily contribute them back.

Is that something you'd be interested in?

**Comment by JonasCir** [comment]
I really like the idea, I am also not a fan of requiring developers to install plugins for simple things like formatting :)

**Comment by JonasCir** [comment]
@MichaelSp @MartinWahnschaffeSymeda 
Any comments on this? :smiley:

**Comment by MichaelSp** [comment]
Yeah that would be nice. What would be even nicer, if the checker reported it's findings in the PR instead of just failing the  build. Not sure if spotless can do that. 
Just did a quick lookup and found a tool that support that https://restyled.io/
But even without that: Better than no check 👍

**Comment by MartinWahnschaffeSymeda** [comment]
Hi. From my understanding when can use spotless to format and check, right?

The check would be something to integrate into the build process, so a PR with bad formatting would fail. I also think that this would be great. The integration of restyled looks even better, but I think we should make sure to have just a single place where the formatting is defined.

The automatic formatting would have to be run by the developer before committing, right? From my point of view this would be a nice alternative to the IDE extensions that automatically do this. Is there a way to somehow integrate this more closely with the IDE, so files are directly formatted?
